aiidalab_widgets_base/process.py

```diff
@@ -220,24 +368,74 @@ def __init__(self, process, **kwargs):
             }
         }
         default_params.update(kwargs)
-        self.previous_calc_id = 0
         self.output = []
-        super(RunningCalcJobOutputWidget, self).__init__(**default_params)
+
+        # Hack to make font monospace. As far as I am aware, currently there are no better ways.
+        display(HTML("<style>textarea, input { font-family: monospace; }</style>"))
+
+        super().__init__(**default_params)
+
+    @observe('calculation')
+    def _change_calculation(self, _=None):
+        """Reset things if the observed calculation has changed."""
+        self.output = []
+        self.value = ''
+
+    def update(self):
+        """Update the displayed output."""
+        if self.calculation is None:
+            return
+
+        output_file_path = None
+        if 'remote_folder' in self.calculation.outputs:
+            output_file_path = os.path.join(self.calculation.outputs.remote_folder.get_remote_path(),
+                                            self.calculation.attributes['output_filename'])
+        if output_file_path and os.path.exists(output_file_path):
+            with open(output_file_path) as fobj:
+                difference = fobj.readlines()[len(self.output):-1]  # Only adding the difference
+                self.output += difference
+                self.value += ''.join(difference)
+
+        # Auto scroll down. Doesn't work in detached mode.
```

Comment:
I assume that's a problem?

Reply:
Well, that is a sort of philosophical question. Essentially, we have to choices here:

1. Have auto scrolling only in the blocked mode
1. Do not have auto-scrolling at all.

To me, the first option does make sense because if you are using the output in the blocking mode, you are kind of keen to follow it. In detached mode, you just want to have the output.

I do agree that the second option is cleaner, but I am not sure if this is too important.

Reply:
I think it's fine to have that slightly inconsistent behavior as long as it is well documented. So make sure to make that clear not only in the in-code comments, but also in the doc-string of the affected function(s). Maybe we can also create an issue for that to keep track of this? I'd classify this as a (tolerated) bug.

Reply:
Added the docstring in 89234fb